pyproject.toml

```
@@ -25,61 +26,60 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = ">=3.7,<3.11"
 click = ">=7.1<=9.0"
-click-default-group = "*"
-colour = "*"
+click-default-group = "^1.2.2"
+colour = "^0.1.5"
 numpy = "^1.9"
-Pillow = "*"
-scipy = "*"
-tqdm = "*"
-pydub = "*"
-pygments = "*"
+Pillow = "^8.4.0"
+scipy = "^1.7.3"
+tqdm = "^4.62.3"
+pydub = "^0.25.1"
 rich = ">=6.0"
 pycairo = "^1.19"
 manimpango = "^0.4.0.post0"
 networkx = "^2.5"
 decorator = "^5.0.7"
-setuptools = "*"
-importlib-metadata = {version = "*", python = "<3.8"}
-grpcio =  { version = "1.33.*", optional = true }
-grpcio-tools = { version = "1.33.*", optional = true }
-watchdog = "*"
+setuptools = "^60.1.0"
+importlib-metadata = { version = "^4.10.0", python = "<3.8" }
+grpcio =  { version = "^1.43.0", optional = true }
+grpcio-tools = { version = "^1.43.0", optional = true }
+watchdog = "^2.1.6"
 jupyterlab = { version = "^3.0", optional = true }
 moderngl = "^5.6.3"
 moderngl-window = "^2.3.0"
 mapbox-earcut = "^0.12.10"
 cloup = "^0.7.0"
-requests = "*"
+requests = "^2.26.0"
```

Pin only the major version? I don't think requests will break API between minor releases.
```suggestion
requests = "^2"
```

---

Shouldn't it then be '"^2.26"'?

---

[Caret requirements](https://python-poetry.org/docs/dependency-specification/#caret-requirements) only pin major version. This means that
- `^2` accepts >= `2.0.0`, < `3.0.0`
- `^2.26` accepts >= `2.26.0`, < `3.0.0`
- `^2.26.2` accepts >= `2.26.2`, < `3.0.0`
I think it should stay the way it is currently.

---

> I think it should stay the way it is currently.

I mean even older versions (from `2.0.0`) should work right?

---

Not necessarily. Manim can rely on some features introduced in later versions (e.g. python 3.10 support). It'd have to be tested and advantages of relaxing constraints that way are minuscule.

---

Also, it doesn't really make sense to relax the restriction here and not the others.

---

> Not necessarily. Manim can rely on some features introduced in later versions (e.g. python 3.10 support). It'd have to be tested and advantages of relaxing constraints that way are minuscule.

Make sense.